javalin/src/main/java/io/javalin/http/servlet/DefaultTasks.kt

```diff
@@ -25,9 +24,19 @@ object DefaultTasks {
             ctx.setRouteRoles(servlet.matchedRoles(ctx, requestUri)) // set roles for the matched handler
             servlet.willMatch(ctx, requestUri)
         }
+        val httpHandlerOrNull by javalinLazy {
+            servlet.router.findHttpHandlerEntries(ctx.method(), requestUri).firstOrNull()
+        }
```

Comment:
Why extract this?

Reply:
There could be multiple matching before Matched handlers as implied by the foreach. I want to avoid the impact of running this multiple times inside the loop.

Reply:
Fair enough